I have done some workaround across implementing ActiveMq to the messagebus so would like to contribute

amitsing89 opened the pull request:
Hi,

I have started once to integrate the activeMq in the messagebus so i would like to add that to the repository, so if you can allow kme to contribute then it would be helpful to add a new component

sibiryakov committed:
hbase states table option

Preetwinder committed:
add message type logging and time for batch

sibiryakov committed:
smaller poll intervals

sibiryakov committed:
more logging in offset-fetcher

sibiryakov committed:
logging contents of e

sibiryakov committed:
change err message and raising kafka log level to INFO

sibiryakov committed:
fix of 'Future' object is not iterable

sibiryakov committed:
simplifying logging configuration

vshlapakov committed:
Send crawl stats to Kafka message bus

Add Scrapy extension to export stats to Kafka
Use mixin class to create new db/sw classes
Extend messagebus class with stats logic
Extend codec logic with encoding/decoding stats
Use export-to-tsdb logic from bigbot-common package
Use single logic for forming SSL params

vshlapakov committed:
Add Mesos task id as a tag for DB worker metrics

vshlapakov committed:
Send backend stats to message bus

Move happybase to dependencies folder
Add get_stats() to distributed backend interface
Provide useful context manager to measure elapsed time
Use the manager to estimate get_stats() efficiency

vshlapakov committed:
lru cache for HBaseState

vshlapakov committed:
Split DB worker to components based on threads

Mv domains blacklist to settings
Rely on NEW_BATCH_DELAY for batchgen backoff

# Conflicts:
#	frontera/worker/db.py

vshlapakov committed:
Fix attribute error related with job_id

sibiryakov committed:
max_request_size is set to 4Mb

vshlapakov committed:
Run only batchgen via thread

vshlapakov committed:
Polish changes and fix minor issues

vshlapakov committed:
Minor code simplification

vshlapakov committed:
Initial job_id must be 0 for db worker

vshlapakov committed:
Smarted backoff for batchgen component

vshlapakov committed:
Do back-off if no requests to schedule

vshlapakov committed:
more precise scoring, encoding from scrapy, hash for request

# Conflicts:
#	frontera/contrib/backends/hbase.py

vshlapakov committed:
No need to setup stats log partitions

vshlapakov committed:
Setting for splitting partitions b/w batchgens

vshlapakov committed:
conflicts resolved

vshlapakov committed:
Config option to pass partitions

vshlapakov committed:
No need new setting

vshlapakov committed:
Decrease log-level for stats producer

vshlapakov committed:
Minor renaming

vshlapakov committed:
Wrap sending message logic

vshlapakov committed:
Update kafkabus producers settings

vshlapakov committed:
Move values to constants

vshlapakov committed:
Non-exhausted generators must be closed manually

vshlapakov committed:
conflicts resolved

vshlapakov committed:
Provide SW states cache hits ratio

vshlapakov committed:
Provide flushed batches count for SW states

vshlapakov committed:
Simplify state stats using with a variable

vshlapakov committed:
Skip links-extracted when reached limit for domain

Write stats about dropped LE events
Add filter_extracted_links method to API

vshlapakov committed:
Provide stats for SW state cache evictions

vshlapakov committed:
Catch and print shutdown errors

sibiryakov committed:
no need to flush on every batch

sibiryakov committed:
create_request method added to manager, etc

sibiryakov committed:
removed domain info logic from hbase.queue, domain mw enabled by default

sibiryakov committed:
enabling domain fingerprint mw by default

sibiryakov committed:
adding cityhash to test reqs

sibiryakov committed:
enabling all tests

sibiryakov committed:
converting row prefix to bytes

sibiryakov committed:
style

sibiryakov committed:
Merge pull request #299 from scrapinghub/headers-etc

hdr and redir persistence + cityhash64 + graceful shutdown fix + kafka SSL

sibiryakov committed:
removed ambiguous redirect logic

sibiryakov committed:
retries budget added in offsets fetch

clarksun committed:
Update hbase.py

imho, in hbase queue's `get_next_requests` method,  params: min_hosts and max_requests_per_host are optional, because this method will check whether they are None, eg line no.225 and no.236, but kwargs.pop will raise Exception if I don't give these two params.

clarksun committed:
maybe miss b?

clarksun committed:
update BCPerHostLimit in cluster example

the `from_worker` method is duplicate , base class `BaseCrawlingStrategy` define `from_worker` already.

sibiryakov committed:
Merge pull request #305 from scrapinghub/logging-improvement

Do not remove existing loggers on configuration

sibiryakov committed:
Merge pull request #302 from scrapinghub/per-type-worker-stats

Add message type logging and time for batch

sibiryakov committed:
Merge pull request #304 from scrapinghub/logging-async-fetcher

Improving errors logging in async offsets fetcher

sibiryakov committed:
Merge pull request #303 from scrapinghub/smaller-kafka-poll-intervals

Smaller kafka client poll intervals

sibiryakov committed:
Merge pull request #306 from clarksun/patch-1

Update hbase.py

sibiryakov committed:
Merge pull request #307 from clarksun/patch-2

maybe miss b?

sibiryakov committed:
Merge pull request #309 from clarksun/patch-4

update BCPerHostLimit in cluster example

sibiryakov committed:
reducing logging verbosity

sibiryakov committed:
Merge pull request #301 from scrapinghub/states-table

HBase states table option

sibiryakov committed:
fix of partitioning in batch-gen

sibiryakov committed:
codecs tests fixed

sibiryakov committed:
other tests fix

sibiryakov committed:
moving stats to worker submodule

sibiryakov committed:
base stats log, zmq stats log

sibiryakov committed:
Merge pull request #313 from scrapinghub/lru-cache-states

LRU cache for HBaseState

sibiryakov committed:
Merge pull request #312 from scrapinghub/stats-log

Stats log

sibiryakov committed:
domain stats in batch gen

sibiryakov committed:
fixes

sibiryakov committed:
hbase refactor, first DomainCache with second gen draft

sibiryakov committed:
converting keys to native str

sibiryakov committed:
set serialization fix

sibiryakov committed:
flush fix and dosctring

sibiryakov committed:
crash fix for py3

sibiryakov committed:
improved error output

sibiryakov committed:
add seeds run mode

# Conflicts:
#	frontera/worker/strategy.py

sibiryakov committed:
overused buffer settings

# Conflicts:
#	frontera/contrib/backends/remote/messagebus.py
#	frontera/core/__init__.py
#	frontera/settings/default_settings.py

sibiryakov committed:
passing args to run()

sibiryakov committed:
removed seeds addition through spider log

# Conflicts:
#	frontera/contrib/backends/remote/codecs/json.py
#	frontera/contrib/backends/remote/codecs/msgpack.py
#	frontera/worker/strategy.py

sibiryakov committed:
codecs test fixed

sibiryakov committed:
renaming add_seeds to read_seeds(file)

sibiryakov committed:
fixed reading of seeds

sibiryakov committed:
s3 dependency

sibiryakov committed:
changing scrapy test website

sibiryakov committed:
ov buffer test fix

sibiryakov committed:
Merge pull request #316 from scrapinghub/misc

more precise scoring, encoding from scrapy, hash for request

sibiryakov committed:
making Kafka mb to support latest kafka-python

sibiryakov committed:
Merge pull request #315 from scrapinghub/kafka-params-up

Kafka-python and kafka params update

sibiryakov committed:
docs

akshayphilar committed:
string formatting corrected in the connection information being logged

sibiryakov committed:
slot for hbase queue partitioning

sibiryakov committed:
db worker test fix

sibiryakov committed:
Merge pull request #314 from scrapinghub/dbw-components-teardown

DBW components teardown

sibiryakov committed:
support of non-iterable argument

sibiryakov committed:
strategy tests fixed

sibiryakov committed:
using unittest

sibiryakov committed:
tests fix

sibiryakov committed:
speeding up serialization

sibiryakov committed:
Merge pull request #327 from scrapinghub/create-req-manager

create_request added to Manager, API changes

sibiryakov committed:
test case

sibiryakov committed:
Merge pull request #324 from scrapinghub/hbase-state-cache

HBase state cache statistics

sibiryakov committed:
Merge pull request #326 from scrapinghub/no-flush-scoring-log

No need to flush scoring log on every batch

sibiryakov committed:
Merge pull request #323 from scrapinghub/generators-pypy

Non-exhausted generators must be closed manually

sibiryakov committed:
Merge pull request #322 from scrapinghub/log-improve-batchgen-sw

Improved logging of batch-gen and SW

sibiryakov committed:
Merge pull request #321 from scrapinghub/part-split-btw-batchgen

Partition split between batch-gen instances

sibiryakov committed:
Merge pull request #320 from scrapinghub/log-connected-host

string formatting corrected in the connection information being logged

sibiryakov committed:
Merge branch 'master' into pre-filter-of-links-ext

sibiryakov committed:
Merge pull request #325 from scrapinghub/pre-filter-of-links-ext

Skip links-extracted when reached limit for domain

sibiryakov committed:
documented None values

sibiryakov committed:
DEFAULT_HBASE_THRIFT_FRAME_SIZE constant

sibiryakov committed:
check for a key presence before deletion

sibiryakov committed:
disabling scrapy test

sibiryakov committed:
codecs tests are fixed

sibiryakov committed:
overusedbuffer test cases are fixed

sibiryakov committed:
more work to codecs fix

sibiryakov committed:
batch_size as argument and bug fix + new test case

sibiryakov committed:
table creation

sibiryakov committed:
removed add_seeds

sibiryakov committed:
crawling strategy test fix

sibiryakov committed:
table presence fix

sibiryakov committed:
add seeds run mode test, sw test fix

sibiryakov committed:
memory backends tests fix

sibiryakov committed:
don't use hbase namespace for testing

sibiryakov committed:
strategy test case fix

sibiryakov committed:
removed add_seeds

sibiryakov committed:
syntax

sibiryakov committed:
Merge pull request #330 from scrapinghub/test-fix2

Tests fixing session

sibiryakov committed:
first draft version

sibiryakov committed:
batched workflow extracted

sibiryakov committed:
more work

sibiryakov committed:
FrontierManager refactoring, more to come

sibiryakov committed:
proper init hierarchy

sibiryakov committed:
import, frontier_manager and strategy fixes

sibiryakov committed:
SW test case fixes

sibiryakov committed:
tests fixed

sibiryakov committed:
tests fixed

sibiryakov committed:
backend tests fixes

sibiryakov committed:
default crawling strategy

sibiryakov committed:
scrapy mws test fix

sibiryakov committed:
dbw test cases fix

sibiryakov committed:
memory backends test fixes

sibiryakov committed:
Redis test case fix

sibiryakov committed:
syntax

sibiryakov committed:
more work on style

sibiryakov committed:
fix of add seeds utility

sibiryakov committed:
fixing distributed run mode

sibiryakov committed:
auto_start/test_mode attrs refactor

sibiryakov committed:
move of BaseCrawlingStrategy, BFS/DFS strategy

sibiryakov committed:
seeds have to be injected using crawling strategy

sibiryakov committed:
new CS guide, and single mode quick start

sibiryakov committed:
removed obsolete test

sibiryakov committed:
new path for a crawling strategy

sibiryakov committed:
style

sibiryakov committed:
DomainMetadata interface

sibiryakov committed:
using DomainMetadata in hbase DomainCache

sibiryakov committed:
SQLA DomainMetadata storage

sibiryakov committed:
moved domain cache test

sibiryakov committed:
DomainCache is instantiated in HBaseBackend now

sibiryakov committed:
removing old memory backends

sibiryakov committed:
removing old sqlalchemy backends

sibiryakov committed:
pointing to existing memory backend

sibiryakov committed:
backends docs update

sibiryakov committed:
local mode instantiation

sibiryakov committed:
error fix

sibiryakov committed:
style

sibiryakov committed:
fix of references in docs

sibiryakov committed:
fixes of single process sqlalchemy tutorial

sibiryakov committed:
Discovery crawling strategy

sibiryakov committed:
removed distributed spiders run mode

sibiryakov committed:
crawling strategy docs

sibiryakov committed:
style mainly

sibiryakov committed:
some mistakes corrected

sibiryakov committed:
putting the code under name=main

sibiryakov committed:
is -> ==

sibiryakov committed:
docstring

sibiryakov committed:
page_error -> request_error

sibiryakov committed:
states tests extraction

sibiryakov committed:
new hbase states test

sibiryakov committed:
fix

sibiryakov committed:
moving states tester

sibiryakov committed:
style

sibiryakov committed:
using fixtures

sibiryakov committed:
sqlalchemy states

sibiryakov committed:
test fix

sibiryakov committed:
table creation

sibiryakov committed:
updated tests

sibiryakov committed:
style

sibiryakov committed:
test fix

sibiryakov committed:
frontera-scrapy test is enabled

sibiryakov committed:
fix warning

sibiryakov committed:
skipping scrapy spider for now

sibiryakov committed:
renaming modules to support py3.7

sibiryakov committed:
to the previous commit

sibiryakov committed:
Merge pull request #331 from scrapinghub/strategy-in-manager

Crawling strategy is moved to FrontierManager

sibiryakov committed:
updated with latest changes

sibiryakov committed:
version bump

sibiryakov committed:
Update README.md

battle tested sentence

sibiryakov committed:
some refactor + devnull producer for stats

sibiryakov committed:
docs + example config update

sibiryakov committed:
Merge pull request #341 from scrapinghub/fix-general-spider

Fix general spider example

sibiryakov committed:
local_mode option

sibiryakov committed:
Merge pull request #342 from scrapinghub/fix-docs

Mention local_mode option in docs

sibiryakov committed:
version bump

icapurro committed:
Remove invalid keyword argument 'sleep' from kafka pull

icapurro committed:
Removed offset unit test

vfcosta committed:
Fix the redis backend

The method flush in RedisState has to define a default value for
force_clear to avoid problems with the frontier manager.

sibiryakov committed:
Merge pull request #347 from vfcosta/fix-redis

Crash on closing redis backend, missing 'force_clear' argument

sibiryakov committed:
Merge pull request #345 from icapurro/fix-kafka-pull

Remove invalid keyword argument 'sleep' from kafka poll

sibiryakov committed:
Merge branch 'master' of https://github.com/scrapinghub/crawl-frontier

sibiryakov committed:
scrapy role, overview update

guillermoap committed:
Update cluster-setup docs

sibiryakov committed:
Merge pull request #351 from guillermoap/update_docs

Update cluster-setup docs

sibiryakov commented:
Hey @amitsing89 , I think you should rebase it to latest master, it seems to me your code is based on some outdated version.

sibiryakov commented:
Currently, it's extremely hard to review because of enormous amount of unrelated changes.

amitsing89 commented:
i will update it and will try to start it again

IAlwaysBeCoding committed:
Docs:  BaseCrawlingStrategy._refresh_states() also takes a non iterable for the requests parameter.

It is important to disclose that `_refresh_states()` also takes a non iterable for the requests parameter. Eventually, `refresh_and_keep()` converts the non iterable object into a non-iterable object contained within a list.

sibiryakov committed:
Update frontera/strategy/__init__.py

Okay that makes sense.

Co-Authored-By: IAlwaysBeCoding <IAlwaysBeCoding@users.noreply.github.com>

sibiryakov committed:
Merge pull request #355 from IAlwaysBeCoding/IAlwaysBeCoding-_refresh_states

Docs:  BaseCrawlingStrategy._refresh_states() also takes a non iterab…

sibiryakov committed:
removing call to absent API in happybase

Commit:
Update after redis update

Use dict of scheduled items rather than a list after redis library update.

sibiryakov committed:
fixed DomainCache crash on init

sibiryakov committed:
Merge pull request #365 from scrapinghub/domain-cache-crash-fix

Fixed DomainCache crash on init for Python 3.6

sibiryakov committed:
Merge pull request #361 from khellan/redis_fix

Update after redis update

sibiryakov committed:
fixing Cache.__init__ signature changes

sibiryakov committed:
more work

sibiryakov committed:
Merge pull request #366 from scrapinghub/test-fix3

Tests fix

Gallaecio commented:
@amitsing89 You should rebase your changes on top of the latest `master` as suggested. Have you used Git rebase before? Have you added the main repository as an `upstream` remote in your local clone of your fork?

sibiryakov committed:
explicit cluster metadata bootstrap in Kafka client

sibiryakov committed:
syntax fixes

sibiryakov committed:
line breaks

sibiryakov committed:
notimplemented->error

Commit:
update models.py

Solving the problem of 'Decode spider logs self.body return none error' https://github.com/scrapinghub/frontera/issues/374

Commit:
Update models.py

Commit:
Update models.py

sibiryakov committed:
Merge pull request #375 from liho00/liho00-patch-1

Update models.py

Commit:
Bump to Python 3.6

Commit:
Use thriftpy2

Commit:
Add Python 3.7

Commit:
Make test_zmq_message_bus() flaky

Commit:
Add .env to gitignore

Commit:
Add badges

Commit:
Add pyversions badge

Commit:
Rename to "python versions"

sibiryakov committed:
Merge pull request #381 from suzil/cleanup/add-badges

Add badges

Commit:
Move thriftpy2 to test_requires

Commit:
Also include thriftpy2 as an hbase requirement

sibiryakov committed:
Merge pull request #379 from suzil/bump/py36

Bump to Python 3.6 + 3.7

a-shkarupin committed:
add assertion error for easier debugging

sibiryakov committed:
Merge pull request #389 from a-shkarupin/assertion_message

add assertion error description for easier debugging

Gallaecio committed:
Port workflows from 0.7.x

Gallaecio committed:
CI publish: fix version tag regexp
